### PR TITLE
Extra extra data #233

### PR DESCRIPTION
**Pull request by r-gochain**
With extra data
image

Without extra data
image

### CODE REVIEW

**Comment by r-gochain** [comment]
@jmank88 could you please?

File: server/models/block.go [hunk 1]
```diff
@@ -12,14 +12,20 @@ type Block struct {
...
 	TotalDifficulty int64     `json:"total_difficulty" bson:"total_difficulty"`
 	Sha3Uncles      string    `json:"sha3_uncles" bson:"sha3_uncles"`
 	ExtraData       string    `json:"extra_data" bson:"extra_data"`
 	// Transactions    []string  `json:"transactions" bson:"transactions"`
+
+	ExtraAuth            bool   `json:"extra_auth" bson:"-"`
+	ExtraVanity          string `json:"extra_vanity" bson:"-"`
+	ExtraHasVote         bool   `json:"extra_has_vote" bson:"-"`
+	ExtraCandidate       string `json:"extra_candidate" bson:"-"`
+	ExtraIsVoterElection bool   `json:"extra_is_voter_election" bson:"-"`
```
Comment (jmank88) [inline server/models/block.go]: Would it be clearer to group these into a struct which is only set when `clique.ExtraHasVote` is true?
```go
type Vote struct {
 Candidate string
 Voter bool
 Auth bool
}
```

File: src/app/scenes/block/block.component.html [hunk 2]
```diff
@@ -61,10 +57,21 @@ <h1>Explore your block</h1>
...
+            <p>Extra Data</p>
+            <b>{{block.extra_vanity }}</b>
+          </li>                          
+        </ul>
+      </li>
+        <ng-container *ngIf="block.extra_has_vote; else empty">
+          <br/>
+              <li>
+                <p>Vote</p>
+                <b>{{block.extra_auth ? "Add" : "Remove"}} {{block.extra_candidate}}</b> to <b>{{block.extra_is_voter_election ? "vote" : "sign"}}</b>
```
Comment (jmank88) [inline src/app/scenes/block/block.component.html]: Reordering to `Add signer 0x1234` might wrap better on mobile.

File: src/app/scenes/block/block.component.html [hunk 3]
```diff
@@ -61,10 +57,21 @@ <h1>Explore your block</h1>
...
         </li>
         <li>
-          <p>Extra Data</p>
-          <b>{{block.extra_data }}</b>
-        </li>
-      </ul>
+        <ul>
+          <li>
+            <p>Extra Data</p>
+            <b>{{block.extra_vanity }}</b>
```
Comment (jmank88) [inline src/app/scenes/block/block.component.html]: Can we use `extra_vanity` on the home page block table as well? And really anywhere else `extra_data` is being used. I'm not sure there is any use for it on the front end in the raw form.

**Pull request merged**